src/components/List/Container.js

```diff
@@ -1,10 +1,54 @@
+import React, { Component } from "react";
 import styled from "styled-components";
+import PropTypes from "prop-types";
+
 import colors from "../../theme/colors";
 import constants from "../../theme/constants";
 
-export default styled.div`
+import { ContainerProvider } from "./Context";
+
+const Container = styled.div`
   width: 100%;
   border: 1px solid ${colors.lightGray};
   border-radius: ${constants.borderRadius.large};
   overflow: hidden;
 `;
+
+class ListContainer extends Component {
+  state = {
+    openIndex: -1
+  };
+
+  onExpandOrCollapse = index => {
+    if (this.state.openIndex) {
+      return this.state.openIndex === index
+        ? this.setState({ openIndex: -1 })
+        : this.setState({ openIndex: index });
+    }
+    return this.setState({ openIndex: index });
+  };
+
+  render() {
+    return (
+      <Container>
+        <ContainerProvider
+          value={{
+            ...this.state
+          }}
+        >
+          {React.Children.map(this.props.children, child =>
+            React.cloneElement(child, {
+              onExpandOrCollapse: this.onExpandOrCollapse
+            })
+          )}
```

Comment:
Rather than cloning the children and passing this prop to each, it would probably be cleaner to have an `onClick` handler on this component or another upper parent component that all expand or collapse events can be delegated to.